crates/flox-rust-sdk/src/models/floxmetav2/mod.rs

```diff
@@ -0,0 +1,39 @@
+use thiserror::Error;
+
+use crate::flox::Flox;
+use crate::providers::git::{GitCommandProvider, GitProvider};
+
+pub const FLOXMETA_DIR_NAME: &str = "meta";
+
+#[derive(Debug)]
+pub struct FloxmetaV2 {
+    pub git: GitCommandProvider,
+}
+
+#[derive(Error, Debug)]
+pub enum FloxmetaV2Error {
+    #[error("todo")]
+    Todo,
+}
+
+impl FloxmetaV2 {
+    pub async fn clone_or_open(
+        flox: &Flox,
+        remote: String,
+        owner: String,
+    ) -> Result<Self, FloxmetaV2Error> {
```

Comment:
@ysndr I haven't really worked through any of the details but wanted to ask what you think about this interface for a ManagedEnvironment to get a floxmeta, since it's a good bit different than what we currently have for floxmeta